libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -44,9 +44,14 @@ type Config struct {
 	// Scope can be either node or cluster.
 	Scope    string `config:"scope"`
 	Resource string `config:"resource"`
+
 	// Unique identifies if this provider enables its templates only when it is elected as leader in a k8s cluster
 	Unique      bool   `config:"unique"`
 	LeaderLease string `config:"leader_lease"`
+	//Parameters to configure election process
+	LeaseDuration time.Duration `config:"leader_leaseduration"`
+	RenewDeadline time.Duration `config:"leader_renewdeadline"`
+	RetryPeriod   time.Duration `config:"leader_retryperiod"`
```

Comment:
These new parameters should also be described in https://github.com/elastic/beats/blob/main/libbeat/docs/shared-autodiscover.asciidoc

Reply:
Yes, I added that as task in the issue

Reply:
wondering if we should align types in the agent https://github.com/elastic/elastic-agent/blob/main/internal/pkg/composable/providers/kubernetesleaderelection/config.go#L17
wdyt?

Reply:
I think it can be misleading that it is different types than agent. Now in metricbeat you need to set it to 15s for example, while in agent to 15. The one looking better is the `time.Duration` type, but changing it in agent would be a breaking change. But as the two projects have differences in general, I would leave it as `time.Duration` here.

Reply:
I agree that it is not great to have different times. For kubernetes secrets provider in elastic agent, we also use `time.Duration`, so maybe it is time to align everything.